src/udisks2monitor.cpp

```diff
@@ -203,6 +203,48 @@ void UDisks2::Monitor::unmount(const QString &devicePath)
     startMountOperation(devicePath, UDISKS2_FILESYSTEM_UNMOUNT, m_blockDevices->objectPath(devicePath), arguments);
 }
 
+void UDisks2::Monitor::setLabel(const QString &devicePath, const QString &label)
+{
+    if (m_blockDevices->find(devicePath)) {
+        QVariantList args;
+        QVariantMap opts;
+        args << opts;
+
+        const QString dbusMethod = UDISKS2_FILESYSTEM_UNMOUNT;
```

Comment:
If the file system is not mounted we should not allow user to rename partition.  
The main idea is to show "Rename" menu item in the context menu for already mounted partition in the Files application or in the Settings=>Storage.
It's possible from UI point of view to allow setting label only when the volume is not mounted in the Settings=>Storage, but it's not possible in Files because unmounted volumes are not listed in Files.
So due to UI limitations in Files (unmounted volumes are not listed) the idea is to show Rename for mounted volumes.

Reply:
> If the file system is not mounted we should not allow user to rename partition.

Ok, although I don't think that makes sense. It might make sense on UI level but in middleware it doesn't IMO.

> The main idea is to show "Rename" menu item in the context menu for already mounted partition in the Files application or in the Settings=>Storage.
> It's possible from UI point of view to allow setting label only when the volume is not mounted in the Settings=>Storage, but it's not possible in Files because unmounted volumes are not listed in Files.

Ok.

> So due to UI limitations in Files (unmounted volumes are not listed) the idea is to show Rename for mounted volumes.

Could you make Files to unmount, rename and mount again the partition instead of limiting it here?

Reply:
Add changes in the UI:
https://os-git.omprussia.ru/jolla/ui-jolla-settings-system/-/merge_requests/41

Reply:
What maybe bothers me slightly is that the mounting/unmounting code gets duplicated here when the Udisks handling is already complex enough here.

What the UI does does not indeed mean that the middleware should behave the same. Here pondering that it could be even warranted to require that the partition is already unmounted when label setting is called, and that remounting is handled by the application side. Even if specific UI somewhere shows rename option only for mounted partitions, it's still possible to do the other steps in the application.

Reply:
Done. Removed unmount and mount operations.

Reply:
I'm happy with this.